include/vara/Feature/ConstraintParser.h

```diff
@@ -0,0 +1,353 @@
+#ifndef VARA_FEATURE_CONSTRAINTPARSER_H
+#define VARA_FEATURE_CONSTRAINTPARSER_H
+
+#include "vara/Feature/Constraint.h"
+#include "vara/Feature/Feature.h"
+
+#include <iostream>
+
+namespace vara::feature {
+
+//===----------------------------------------------------------------------===//
+//                               ConstraintToken
+//===----------------------------------------------------------------------===//
+
+class ConstraintToken {
+public:
+  using Precedence = int;
+
+  enum class ConstraintTokenKind {
+    ERROR,
+    END_OF_FILE,
+    WHITESPACE,
+    IDENTIFIER,
+    NUMBER,
+    L_PAR,
+    R_PAR,
+    NOT,
+    OR,
+    AND,
+    EQUAL,
+    NOT_EQUAL,
+    LESS,
+    GREATER,
+    IMPLIES,
+    EQUIVALENT,
+    PLUS,
+    MINUS,
+    STAR,
+    NEG
+  };
+
+  ConstraintToken(ConstraintTokenKind Kind) : Kind(Kind) {}
+  ConstraintToken(ConstraintTokenKind Kind, std::string Value)
+      : Kind(Kind), Value(Value) {}
+  ConstraintToken(const ConstraintToken &) = delete;
+  ConstraintToken &operator=(ConstraintToken &) = delete;
+  ConstraintToken(ConstraintToken &&) = delete;
+  ConstraintToken &operator=(ConstraintToken &&) = delete;
+  virtual ~ConstraintToken() = default;
+
+  [[nodiscard]] ConstraintTokenKind getKind() const { return Kind; };
+
+  [[nodiscard]] std::optional<std::string> getValue() { return Value; }
+
+  [[nodiscard]] Precedence calcPrecedence() const {
```

Comment:
Precedence changes as discussed last Friday, switching default, what happens to brackets etc.